content/docs/rfcs/37/README.md

```diff
@@ -0,0 +1,151 @@
+
+Currently, it is not possible to mark a Noise session as stale when `session-id` is lost or gets corrupted in users' devices.
+However, since `session-id` is shared between Alice and Bob, 
+one party MAY decide to mark a Noise session as stale if no message from the other end was received within a certain fixed time window. 
+
+The above mechanism allows a Noise session to be marked as stale either privately or publicly, 
+depending if `Hash(session-id)` is sent on `/wakunoise/1/<ct-id>/proto` to the other party in encrypted form or not, respectively. 
+
+When a Noise session is publicly marked as stale, 
+network peers MAY discard all [stored](https://rfc.vac.dev/spec/13/) messages addressed to the content topic `/wakunoise/1/<ct-id>/proto`. 
+In this the case and in order for parties to retrieve any eventually delayed message, 
+peers SHOULD wait a fixed amount of time before discarding stored messages corresponding to a stale Noise session. 
+
+A stale Noise session cannot be directly marked as active
+and parties are required to instantiate a new Noise session if they wish to communicate again. 
+
+However, parties can optionally include the `session-id` corresponding to a stale Noise session in the [prologue information](https://noiseprotocol.org/noise.html#prologue) employed in the Noise handshake they execute to instantiate their new Noise session. 
+This effectively emulates a mechanism to _"re-activate"_ a stale Noise session by binding it to a newly created active Noise session.
+
+In order to reduce users' metadata leakage, it is desirable (as suggested in [35/WAKU2-NOISE](https://rfc.vac.dev/spec/35/#after-handshake)) that content topics used for communications change every time a new message is exchanged.
+This can be easily realized by employing a key derivation function to compute a new `session-id` from the previously employed one (e.g. `session-id = HKDF(prev-session-id)`),
+while keeping the Inbound/outbound Cipher States, the content topic derivation mechanism and the stale mechanism the same as above.
+In this case, when one party sends **and** receives at least one message, 
+he SHALL publicly mark as stale all Noise sessions relative to messages exchanged before the earlier of these two send/receive events.
+
+# Multi-Device support
+
+Alice and Bob might possess one or more devices (e.g. laptops, smartphones, etc.) they wish to use to communicate.
+In the following we assume Alice and Bob to possess $N$ and $M$ devices, respectively.
+
+Since a Noise session contains cryptographic material required to encrypt and decrypt messages exchanged on a pre-defined content topic derived from a `session-id`, 
+messages should be encrypted and decrypted within the Noise session instantiated between the currently-in-use sender's and receiver's device.
+
+This is achieved through two main supported session management mechanisms that we called `N11M` and `NM`, respectively.
```

Comment:
What do these names `N11M` and `NM` stand for? are they abbreviations of something else?

Reply:
I agree that the names are not the best possible. These names sketch the Noise session graph layout: 
- In NM all Alice's N devices have a session with all Bob's M devices, that is: each Alice's N devices <-> each Bob's M devices.
- In N11M, there is a 1-to-1 session shared among all N and all M Alice's and Bob's devices, that is: (Alice N devices) <-> (Bob M devices). 

I thought that pictures would help understanding the reason for such names, but by being just names I didn't want to go into the details of their meaning.

Reply:
I actually like the names. They are kind of in line with noise protocol names.

Reply:
> In NM all Alice's N devices have a session with all Bob's M devices, that is: each Alice's N devices <-> each Bob's M devices.
In N11M, there is a 1-to-1 session shared among all N and all M Alice's and Bob's devices, that is: (Alice N devices) <-> (Bob M devices).

Thanks for the clarification!